src/gui/sdl_mapper.cpp

```diff
@@ -1754,41 +1764,135 @@ static void change_action_text(const char* text,Bit8u col) {
 	bind_but.action->SetColor(col);
 }
 
+static std::string humanize_mod_name(const CBindList &binds, const std::string &fallback)
+{
+	auto trim_pfx = [](const std::string &pfx, const std::string &input) {
+		if (starts_with(pfx, input))
+			return input.c_str() + pfx.size();
+		else
+			return input.c_str();
+	};
+
+	const auto binds_num = binds.size();
+
+	// We have a single bind, just use it
+	if (binds_num == 1)
+		return binds.front()->GetBindName();
+
+	// Avoid prefix, e.g. "Left Alt" and "Right Alt" -> "Alt"
+	if (binds_num == 2) {
+		const std::string new_pfx = (fallback.empty() ? "" : fallback + ": ");
+		const std::string name_1 = binds.front()->GetBindName();
+		const std::string name_2 = binds.back()->GetBindName();
+		std::string trimmed_1 = trim_pfx("Right ", name_1);
+		std::string trimmed_2 = trim_pfx("Left ", name_2);
+		if (trimmed_1 == trimmed_2)
+			return new_pfx + trimmed_1;
+		trimmed_1 = trim_pfx("Left ", name_1);
+		trimmed_2 = trim_pfx("Right ", name_2);
+		if (trimmed_1 == trimmed_2)
+			return new_pfx + trimmed_1;
+	}
+
+	return fallback;
+}
+
+static void SetActiveBind(CBind *new_active_bind)
```

Comment:
A description of everything this function does would help. It looks like it does logical internal stuff as well as update the mapper's GUI?

Or maybe it could be split into two functions: one that does the internal logical update and the other does the GUI update; although maybe they share variables so that's why it turned into one function.

Reply:
It stores pointer to the selected "bind" object in global variable and updates GUI, nothing else.

All logic I added in here exists only for the purpose of creating a nice key combination description.

Reply:
This would also be a good comment to add; nothing major but helpful to describe what each section is doing.

Reply:
I split SetActiveBind into two functions; new function name clearly communicates that it updates the UI only :)